Update platform onboarding checklist with current tooling and docs

stiehlrod opened the pull request:
- Fix outdated tool access URL format
- Add Sentry deprecation note (being replaced by DataDog)
- Emphasize DataDog as primary monitoring platform
- Add GitHub Copilot onboarding reference
- Fix vets-gov-write team name (was vets.gov-write)
- Add Argo CD context (GitOps deployment tool)
- Add Test User Dashboard reference
- Add Developer onboarding section with Platform docs, local setup guides, and PII Guidelines

stiehlrod committed:
Update platform onboarding checklist with current tooling and docs

- Fix outdated tool access URL format
- Add Sentry deprecation note (being replaced by DataDog)
- Emphasize DataDog as primary monitoring platform
- Add GitHub Copilot onboarding reference
- Fix vets-gov-write team name (was vets.gov-write)
- Add Argo CD context (GitOps deployment tool)
- Add Test User Dashboard reference
- Add Developer onboarding section with Platform docs, local setup guides, and PII Guidelines

rmtolmach commented on .github/ISSUE_TEMPLATE/platform-onboarding-checklist.md:
```diff
@@ -67,17 +67,26 @@ RA - Remote Access_
-  - [ ] Even though you're Platform and not VFS, fill out the [New VFS Team Member Template](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?template=New-VFS-Team-Member.yml). By filling that out, you'll get added to the `vets.gov-write` GitHub Team which grants you write access to most repos in the org.
+  - [ ] Even though you're Platform and not VFS, fill out the [New VFS Team Member Template](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?template=New-VFS-Team-Member.yml). By filling that out, you'll get added to the `vets-gov-write` GitHub Team which grants you write access to most repos in the org.
   - [ ] Make sure you have access to the [devops repo](https://github.com/department-of-veterans-affairs/devops)
+  - [ ] [GitHub Copilot](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?template=github-copilot-onboarding.yml) - _AI pair programming assistant, available after GitHub org access_
```
This is linked on the "Request access to tools" page https://depo-platform-documentation.scrollhelp.site/getting-started/request-access-to-tools#Requestaccesstotools-GettingCopilotaccess. Not sure if we need it here as well. But not opposed to it.

jweissman reviewed (approved):
verifying i'm able to approve things (but it does lgtm outside of the issues @rmtolmach is pointing out!)

stiehlrod committed:
Address PR feedback on onboarding checklist

- Revert vets.gov-write (display name uses period, not hyphen)
- Move Test User Dashboard up with other VA Network tools, add Legacy note
- Update PII link to research-design page that lists all PII

rmtolmach committed:
add question mark

rmtolmach reviewed (approved):
LGTM!
